.github/workflows/doc.nuxeo.com.yml

```
@@ -0,0 +1,73 @@
+name: Build and Deploy doc.nuxeo.com
+
+on:
+  push:
+    branches:
+      - master
+      - development
+
+jobs:
+  set-environment:
```

The job ID do not really represent its purpose. Something like `documentation`  is more representative IMHO.
You may also want to give the job a more readable name.

```suggestion
  documentation_ci:
    name: Build & Publish documentation to ${{ github.ref_name == 'master' && 'production' || github.ref_name }}
```